proposed/http-client/http-client.md

```diff
@@ -21,9 +21,9 @@ implementations. This would make libraries more stable since the reduced number
 dependencies and the likelihood to get in version conflicts is reduced.
 
 The second goal is that all HTTP clients should follow the [Liskov substitutions principle][Liskov].
-This means that all clients should act the same when sending a request. By default a HTTP client
-should not follow redirect nor throw exceptions on HTTP responses with status 4xx or 5xx.
-
+This means that all clients should act the same when sending a request. A consuming
+library SHOULD expect a HTTP client to NOT follow redirects NOR throw exceptions based
+on HTTP response statuses.
```

Comment:
Technically it shouldn't matter to a consuming code if the client followed redirections because it will receive a response anyway.

On the other hand if the client does not follow redirections, the consuming code needs to deal with it, which can actually be a valid use case too. For example: based on the redirected host new Authorization header needs to be added, etc.

I think it's okay not to follow redirections by default, because most of the cases the person who writes the consuming code also knows if it should expect a redirection or not.